main.rb

```diff
@@ -0,0 +1,470 @@
+
+    # if we reach this step and all elements are thruty, we exit true.
+  end
+
+  def no_block_count(element_check, block, arr_i)
+    element_check += 1 if (!arr_i.nil? || arr_i != false) && !block
+    element_check
+  end
+
+  # my_any? method definition
+  def my_any?(param = String)
+    arr = self
+    any_h = { true_block_elements: 0, true_elements: 0, t_rxp: 0, true_pattern: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if block_given?
+        any_h[:true_block_elements] += 1 if yield arr[i]
+      end
+      any_h[:true_block_elements] = no_block_count(any_h[:true_block_elements], block_given?, arr[i])
+      any_h[:t_rxp], any_h[:true_pattern] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:true_pattern], false)
+    end
+    any_h[:true_block_elements] >= 1 || any_h[:t_rxp] >= 1 || any_h[:true_pattern] >= 1
+  end
+
+  def any_element_true(len, fbe, fel, freg, fpa)
+    # method used inside my_none
+    fbe == len || fel == len || freg == len || fpa == len
+  end
+
+  # my_none? method definition
+  def my_none?(param = '')
+    arr = self
+    any_h = { f_b_e: 0, f_e: 0, f_rxp: 0, f_p: 0, who: true }
+
+    0.upto(arr.length - 1) do |i|
+      if param == ''
+        if block_given?
+          any_h[:f_b_e] += 1 unless yield arr[i]
+        elsif arr[i].nil? || arr[i] == false
+          any_h[:f_e] += 1
+        end
+      end
+      any_h[:f_rxp], any_h[:f_p] = param_reg(param, arr[i], any_h[:f_rxp], any_h[:f_p], any_h[:who])
+    end
+    any_element_true(arr.length, any_h[:f_b_e], any_h[:f_e], any_h[:f_rxp], any_h[:f_p])
+  end
+
+  # my_map method definition (updated after point 10 of assigment)
+  def my_map(*my_proc)
+    caller = self
+    return to_enum :my_map_proc unless block_given? || my_proc[0].class == Proc
+
+    back = []
+    caller.my_each do |value|
+      if my_proc[0].is_a? Proc
+        back.push my_proc[0].call value
+      elsif block_given?
+        back.push(yield value)
+      end
+    end
+    back
+  end
+
+  # my_count method definition
+  def my_count(param = '')
+    caller = self
+    back = 0
+    caller.my_each do |value|
+      if block_given?
+        back += 1 if yield value
+      elsif param != '' && !block_given?
+        back += 1 if value == param
+      else
+        back += 1
+      end
+    end
+    back
+  end
+
+  def conditions(con1, con2, con3)
+    con1 && con2 && con3
+  end
+
+  # my_inject method definition
+  def my_inject(*param)
+    caller = self
+    i = 0
+    total = []
+    caller.my_each do |value|
+      if conditions(i.zero?, !param.nil?, param[0].is_a?(Numeric))
+        total[0] = param[0]
+      elsif conditions(i.zero?, block_given?, value.is_a?(Numeric))
+        total[0] = value - value
+      elsif conditions(i.zero?, block_given?, !value.is_a?(Numeric))
+        total[0] = value.clear
+      end
+      total[i + 1] = yield total[i], value
+      i += 1
+    end
+    total[-1]
+  end
+end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
array = Array.new(100) { rand(0...9) }
array.my_inject(:+) == array.inject(:+) #true
```

Reply:
This method is now working on my end
irb(main):990:0> array = Array.new(100) { rand(0...9) }
=> [7, 1, 7, 3, 1, 0, 8, 6, 2, 6, 1, 4, 1, 3, 8, 7, 1, 6, 3, 5, 4, 0, 6, 7, 8, 0, 7, 7, 7, 8, 6, 6, 5, 7, 4, 6, 4, 0, 0, 8, 3, 3, 1, 0, 0, 5, 4, 8, 8, 5, 7, 0, 4, 2, 0, 1, 5, 4, 1, 7, 4, 1, 1, 8, 1, 5, 6, 8, 4, 4, 4, 6, 2, 2, 8, 6, 5, 4, 1, 5, 4, 7, 2, 3, 7, 2, 1, 6, 6, 4, 5, 4, 7, 3, 5, 4, 3, 0, 5, 6]
irb(main):991:0> array.my_inject(:+)
=> 417
irb(main):992:0> array.inject(:+)
=> 417
irb(main):993:0> array.my_inject(:+) == array.inject(:+) 
=> true
irb(main):994:0>